File: 383-ransom-note/canConstruct.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

# from collections import Counter
import collections

class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        count = collections.Counter(magazine)
        for c in ransomNote:
            count[c] -= 1
            if count[c] < 0:
                return False
        return True

s = Solution()
print(s.canConstruct("aa", "aab"))

File: 383-ransom-note/canConstruct.cpp
#include <iostream>
#include <unordered_map>

using namespace std;
bool canConstruct(string ransomNote, string magazine) {
    unordered_map<char, int> myMap;
    for (auto c : magazine) myMap[c]++;
    for (auto c : ransomNote) {
        myMap[c]--;
        if (myMap[c] < 0) return false;
    }
    return true;
}

int main() {
    cout << canConstruct("aa", "aab") << endl;
    return 0;
}
